arangod/Cluster/ClusterInfo.h

```diff
@@ -287,6 +287,54 @@ class CollectionInfoCurrent {
                              // underpins the data presented in this object
 };
 
+
+class AnalyzerModificationTransaction {
+public:
+  AnalyzerModificationTransaction(DatabaseID const& database, ClusterInfo* ci, bool cleanup)
+    : _clusterInfo(ci), _database(database), _cleanupTransaction(cleanup) {
+    TRI_ASSERT(_clusterInfo);
+  }
+
+  AnalyzerModificationTransaction(AnalyzerModificationTransaction const&) = delete;
+  AnalyzerModificationTransaction& operator=(AnalyzerModificationTransaction const&) = delete;
+
+  ~AnalyzerModificationTransaction() {
+    abort();
```

Comment:
we have to ensure `abort()` is marked `noexcept` or suppress exceptions in destructor